technical-guidelines.md

```diff
@@ -0,0 +1,120 @@
+| static analysis                                | [SonarCloud][sonarcloud], language-specific tools (SpotBugs, eslint)                                                                      | recommended                                                 |
+
+## Semantic Versioning and 1.0 Releases
+
+We require release artifacts to adhere to [semantic versioning](https://semver.org/) except in specific cases approved by the [Technical Steering Committee](https://github.com/open-feature/community/blob/main/community-members.md#technical-committee).
+1.0 releases must satisfy the [repository requirements](#repository-requirements) above.
+If the artifact is an SDK implementing the OpenFeature specification, it must also conform to a version of the OpenFeature specification not less than 2 minor versions behind the latest (ex: if the latest OpenFeature specification is `0.8.1`, then an implementation conforming to `0.6.0` is a candidate for `1.0` release, while an implementation conforming only to `0.5.0` is not).
+This policy may be subject to change with the release of the OpenFeature specification version `1.0.0`.
+
+### Breaking Changes to Major Versions
+
+Efforts should be made to avoid breaking changes in `1.0+` artifacts.
+If such changes are necessary, please consider the following:
+
+* How important is the breaking change? Is there a non-breaking alternative?
+* Have the changes in question been marked as deprecated for some time?
+  * If possible, deprecate the functionality and give users some time to prepare for the changes in advance.
+* How difficult would it be to backport emergency fixes to the previous version (enterprises on the old platform will almost certainly not be able to upgrade immediately, so backports for security fixes or severe bugs must be supported). 
+  * It's highly recommended to create a `v(-1)` branch before introducing breaking changes, to support the release of such fixes.
+    * For example, if you are releasing a `2.0.0`, maintain a `v1` branch.
+    * [Release Please][release-please] can make maintenance and release of artifacts from such a branch easy; simply have the release-please action run on the branch, and target that branch for its PRs.
+
+Please consult with a member of the [TC](https://github.com/open-feature/community/blob/main/community-members.md#technical-committee) before making breaking changes to a `1.0+` component.
+
+## Platform Support
+
+OpenFeature maintains SDKs and integrations targeting various platforms, runtimes, and frameworks.
+This section describes our approach to supporting, and deprecating support for these.
+
+### Platform Support Recommendations
+
+* Efforts should be made to support the usage of OpenFeature libraries on all officially supported (non-EOL) platforms, at a minimum.
```

Comment:
can we link "all officially supported platforms" here to wherever we define that?

Reply:
I had trouble wording this line. What I meant was all supported versions of the platform the SDK targets, ie: for node, the current supported versions are [18/20](https://endoflife.date/nodejs). The idea is if you are writing a C# SDK, you should support the currently officially supported versions of C#, etc.

Reply:
guess, you could say officially supported/maintained versions up to two versions or something?

Reply:
I think we may better need to define what a platform is in respect to an SDK. We are talking about language, frameworks, and virtual machines. Maybe sometimes we will be talking about OS as well (iOS, Android, C++). Go for Go we are talking about the Go version, not Windows 11, for instance. For some SDKs you my need 1 more level. The react SDK would likely have version constraints on react, browsers, and maybe node.

Reply:
Maybe we can provide some examples and then defer, to some extent, what a platform is to the SDK.

The platforms being targeted also evolve and that may provide some leeway. You may make a server SDK and it supports node. Eventually it supports deno, bun, edge environments, and incrementally has more platform requirements.

Reply:
I was also thinking examples for `runtimes`, `language versions` or `OS`s` could be helping this point.

Reply:
I've updated this to `Efforts should be made to support the usage of OpenFeature libraries on all officially supported (non-EOL) platforms for the library in question (i.e.: a library for Node.js should support all officially supported versions of Node.js).`

Reply:
> Maybe we can provide some examples and then defer, to some extent, what a platform is to the SDK.

> The platforms being targeted also evolve and that may provide some leeway. You may make a server SDK and it supports node. Eventually it supports deno, bun, edge environments, and incrementally has more platform requirements.

Good points.

Reply:
@kinyoklion To the preceding section, I've added:

```
For the purposes of this document, a "platform" constitutes an assumed underlying runtime dependency.
This includes, but is not limited to: languages, operating systems, dependency managers, virtual machines and software libraries.
```